### ISSUE TITLE
Build error with Clang

### ISSUE THREAD

**Issue by eschnett** [LLNL/Silo#302]
I am building Silo 4.11 with Clang 15.0.7, and I encounter this error:

```
  >> 667    silo_hdf5.c:1870:13: error: incompatible integer to pointer conversion passing 'char' to parameter of type 'const char
            *'; take the address with & [-Wint-conversion]
     668                DB_OBJ_CASE(DB_CURVE, DBcurve_mt, npts?1:1, yvarname)
     669                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     670    silo_hdf5.c:1813:36: note: expanded from macro 'DB_OBJ_CASE'
     671                (*dsnames)[i] = strdup(m.MEMNAME[i]);                \
     672                                ~~~~~~~^~~~~~~~~~~~~
     673    ./../silo/silo_private.h:939:35: note: expanded from macro 'strdup'
```

The problem is that the macro expands to `m.yvarname`, and this is a string, not an array of strings as expected in this macro. Thus `m.yvarname[i]` is a character, not a pointer to a string.

**Comment by patrickb314** [comment]
I get the same problem building with Cray compilers (which are also clang based).

**Comment by markstock** [comment]
I'm working around this by making a copy of the macro just for the `DB_CURVE` entry and removing the `[i]` at the end of `MEMNAME`. The rest of the build progresses well. Using AMD clang 15.0.0.

**Comment by markcmiller86** [comment]
Sorry for delay in responding. This has been corrected on `4.11RC` in [this commit](https://github.com/LLNL/Silo/commit/be29ddf0352bc8e5a7eecc8772a3acb64dfde18c)

**Comment by markcmiller86** [comment]
You can download a patch by...
 
https://github.com/LLNL/Silo/commit/be29ddf0352bc8e5a7eecc8772a3acb64dfde18c.patch
 
So, do this...
 
1. `cd` to wherever you have silo-4.11 source tree top dir
1. `wget https://github.com/LLNL/Silo/commit/be29ddf0352bc8e5a7eecc8772a3acb64dfde18c.patch`
1. `patch src/hdf5_drv/silo_hdf5.c be29ddf0352bc8e5a7eecc8772a3acb64dfde18c.patch`
